src/main/java/org/broadinstitute/hellbender/tools/pon/allelic/AllelicPanelOfNormalsCreator.java

```diff
@@ -0,0 +1,82 @@
+    private static final Logger logger = LogManager.getLogger(AllelicPanelOfNormalsCreator.class);
+
+    private final List<File> pulldownFiles;
+
+    /**
+     * Constuctor given pulldown files for all normals in the panel.
+     * @param pulldownFiles pulldown files for all normals in the panel
+     */
+    public AllelicPanelOfNormalsCreator(final List<File> pulldownFiles) {
+        Utils.nonNull(pulldownFiles, "List of pulldown files cannot be null.");
+        ParamUtils.isPositive(pulldownFiles.size(), "List of pulldown files should contain at least one file.");
+        pulldownFiles.stream().forEach(Utils::regularReadableUserFile);
+        this.pulldownFiles = new ArrayList<>(pulldownFiles);
+    }
+
+    /**
+     * Creates an {@link AllelicPanelOfNormals} given a site-frequency threshold;
+     * sites appearing in strictly less than this fraction of samples will not be included in the panel of normals.
+     * @param siteFrequency site-frequency threshold
+     * @return              an {@link AllelicPanelOfNormals} containing sites above the site-frequency threshold
+     */
+    public AllelicPanelOfNormals create(final double siteFrequency) {
+        logger.info("Creating allelic panel of normals...");
+        final Map<SimpleInterval, MutableInt> numberOfSamplesMap = new HashMap<>(); //used to filter on frequency
+        final Map<SimpleInterval, AllelicCount> totalCountsMap = new HashMap<>();   //store only the total counts (smaller memory footprint)
+        int pulldownFileCounter = 1;
+        final int totalNumberOfSamples = pulldownFiles.size();
+        for (final File pulldownFile : pulldownFiles) {
+            logger.info("Processing pulldown file " + pulldownFileCounter++ + "/" + totalNumberOfSamples + " (" + pulldownFile + ")...");
+            final AllelicCountCollection pulldownCounts = new AllelicCountCollection(pulldownFile);
+            for (final AllelicCount count : pulldownCounts.getCounts()) {
+                //update the sum of ref and alt counts at each site
+                final SimpleInterval site = count.getInterval();
+                final AllelicCount currentCountAtSite = totalCountsMap.getOrDefault(site, new AllelicCount(site, 0, 0));
+                final AllelicCount updatedCountAtSite = new AllelicCount(
+                        site,
+                        currentCountAtSite.getRefReadCount() + count.getRefReadCount(),
+                        currentCountAtSite.getAltReadCount() + count.getAltReadCount());
+                totalCountsMap.put(site, updatedCountAtSite);
+                //update the number of samples seen possessing each site
+                final MutableInt numberOfSamplesAtSite = numberOfSamplesMap.get(site);
+                if (numberOfSamplesAtSite == null) {
+                    numberOfSamplesMap.put(site, new MutableInt(1));
+                } else {
+                    numberOfSamplesAtSite.increment();
+                }
```

Comment:
I'm not actually sure!  In general, I wouldn't say we try to favor particular implementations of specific algorithms.  As long as your implementation is 1) working and correct, 2) understandable and maintainable, 3) not terribly inefficient, we'll probably allow it (although the passing threshold for each is up to the PR reviewer).  Of course, if certain bits of code are crucial to performance then we'll optimize where necessary, but that can happen in subsequent PRs.